cmd/thanos/query.go

```diff
@@ -304,8 +305,14 @@ func runQuery(
 		)
 		proxy            = store.NewProxyStore(logger, reg, stores.Get, component.Query, selectorLset, storeResponseTimeout)
 		rulesProxy       = rules.NewProxy(logger, stores.GetRulesClients)
-		queryableCreator = query.NewQueryableCreator(logger, reg, proxy, maxConcurrentSelects, queryTimeout)
-		engine           = promql.NewEngine(
+		queryableCreator = query.NewQueryableCreator(
+			logger,
+			extprom.WrapRegistererWithPrefix("thanos_query_concurrent_selects_", reg),
```

Comment:
Right now, this registry only used for concurrent selects. However, it could be used for others as well. I think we should wrap the registry this early.

Reply:
Good point. I'll think about it.

Reply:
done